src/components/NewCommentForm.tsx

```diff
@@ -9,63 +16,79 @@ type Props = {
 export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const [submitting, setSubmitting] = useState(false);
 
-  const [errors, setErrors] = useState({
-    name: false,
-    email: false,
-    body: false,
-  });
-
-  const [{ name, email, body }, setValues] = useState({
-    name: '',
-    email: '',
-    body: '',
+  const { errors } = useAppSelector(state => state.newComment);
+  const { name, email, body } = useAppSelector(state => {
+    return state.newComment.newComment;
   });
+  const dispatch = useAppDispatch();
 
   const clearForm = () => {
-    setValues({
+    dispatch(setNewComment({
       name: '',
       email: '',
       body: '',
-    });
+    }));
 
-    setErrors({
+    dispatch(setErrors({
       name: false,
       email: false,
       body: false,
-    });
+    }));
   };
 
   const handleChange = (
     event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
   ) => {
     const { name: field, value } = event.target;
 
-    setValues(current => ({ ...current, [field]: value }));
-    setErrors(current => ({ ...current, [field]: false }));
+    const newErrosData: ErrorForForm = {
```

Comment:
The variable 'newErrosData' has a typo. Consider renaming it to 'newErrorsData'.